libyul/backends/wasm/WasmDialect.cpp

```diff
@@ -38,32 +38,46 @@ WasmDialect::WasmDialect()
 			"add",
 			"sub",
 			"mul",
+			// TODO: div_s
 			"div_u",
+			// TODO: rem_s
 			"rem_u",
 			"and",
 			"or",
 			"xor",
 			"shl",
+			// TODO: shr_s
 			"shr_u",
+			// TODO: rotl
+			// TODO: rotr
 		})
 			addFunction(t.str() + "." + name, {t, t}, {t});
 
 	for (auto t: types)
 		for (auto const& name: {
 			"eq",
 			"ne",
+			// TODO: lt_s
 			"lt_u",
+			// TODO: gt_s
 			"gt_u",
+			// TODO: le_s
 			"le_u",
+			// TODO: ge_s
 			"ge_u"
 		})
 			addFunction(t.str() + "." + name, {t, t}, {i32});
 
 	addFunction("i32.eqz", {i32}, {i32});
 	addFunction("i64.eqz", {i64}, {i32});
 
-	addFunction("i32.clz", {i32}, {i32});
-	addFunction("i64.clz", {i64}, {i64});
+	for (auto t: types)
+		for (auto const& name: {
+			"clz",
+			"ctz",
+			"popcnt",
+		})
```

Comment:
Why is this done via loop only for these 3?

Reply:
`eqz` always returns i32, while these return the same type as the input.